contracts/MultiSender.sol

```diff
@@ -6,69 +6,122 @@ import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
 import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
 import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
 
 // Use SafeERC20 library for IERC20Upgradeable
 using SafeERC20 for IERC20Upgradeable;
 
 // Define the contract
-contract MultiSender is Initializable {
+contract MultiSender is Initializable, ReentrancyGuardUpgradeable {
     // Initialize the contract
-    function initialize() public initializer {}
+    function initialize() public initializer {
+        __ReentrancyGuard_init();
+    }
 
     // Event definitions
     event SendETH(address[] recipients, uint256[] amounts);
     event SendERC20(address token, address[] recipients, uint256[] amounts);
 
     // Function to send ETH to multiple recipients
-    function sendETH(address[] calldata _recipients, uint256[] calldata _amounts) external payable {
+    function sendETH(
+        address[] calldata _recipients,
+        uint256[] calldata _amounts
+    ) external payable nonReentrant {
         // Check if the lengths of recipients and amounts arrays are equal
         require(_recipients.length == _amounts.length, "Must have the same length");
 
         // Calculate the total amount to be sent
-        uint256 totalAmount = 0;
+        uint256 _totalAmount = 0;
         for (uint256 i = 0; i < _amounts.length; i++) {
-            totalAmount += _amounts[i];
+            _totalAmount += _amounts[i];
         }
 
+        // Check if the total amount is greater than zero
+        require(_totalAmount > 0, "Total transfer amount is zero");
+
         // Check if the sender has enough ETH
-        require(msg.value == totalAmount, "Unequal transfer amount");
+        require(msg.value == _totalAmount, "Unequal transfer amount");
```

Comment:
Thank you, @bayram98 
I think we can remove this checking, because of having other check. I think those checking give us same result. 
 require(finalBalance == initialBalance, "Balance mismatch after transfers");